### PR TITLE
Add mixed server pre-req 

### PR DESCRIPTION
**Pull request by sabuchan07**
Cherry-pick into:

* [x] Foreman 3.3
* [x] Foreman 3.2
* [ ] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

### CODE REVIEW

**Comment by maximiliankolb** [comment]
Hi @sabuchan07 Please rebase your PR against master so that you have exactly one commit on your branch (unless you want two separate changes which is perfectly fine!), and have a look at https://cbea.ms/git-commit/#seven-rules. Please ping me when you want me to review this.

**Comment by sabuchan07** [comment]
Hi, @maximiliankolb !  Thank you for the link to the commit message instructions. Very helpful!  Also, forgive my ignorance, but I rebased in my command line, but nothing seems to have changed here.  Is this something I also need to do in the GitHub UI?  Thank you!

**Comment by maximiliankolb** [comment]
> 

I assume that you need to fetch and pull from the master branch from Github into your local master branch first, then rebase again.

**Comment by maximiliankolb** [comment]
Since you definitely have to rebase to master due to conflicts, you can decide for yourself whether to squash your commits while rebasing or not. I can, when merging, squash your commits and also adjust the commit message.

**Comment by melcorr** [comment]
Your local setup doesn't look correct if it is dragging in all this detritus @sabuchan07 
You shouldn't be merging locally as per [15a2956](https://github.com/theforeman/foreman-documentation/pull/1293/commits/15a2956afe18f430d429aa5c58141a858f07c731)
Your commit message also make no sense and we really need them to have relevant message as per https://cbea.ms/git-commit/ - this is included in the setup instructions. 
This looks messy and merging this mess would add mess to our repo.

File: guides/common/modules/proc_configuring-repositories.adoc [hunk 1]
```diff
@@ -2,14 +2,16 @@
...
+Choose from the available list which operating system and version you are installing on:
 
 ifdef::foreman-el,katello[]
 * xref:#repositories-centos-8[CentOS 8]
 * xref:#repositories-centos-7[CentOS 7]
 endif::[]
 ifdef::foreman-el,katello,satellite[]
-* xref:#repositories-rhel-7[{RHEL} 7]
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
+//* xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-repositories.adoc:14]: We generally don't add comments like this. Is there anything that readers of the source code should know about el8?

Please either remove it or change it to something like `// _a very imporant/interesting note to maintainers_`.

File: guides/common/modules/ref_system-requirements.adoc [hunk 2]
```diff
@@ -41,6 +41,10 @@ endif::[]
 * A system umask of 0022
 * Full forward and reverse DNS resolution using a fully-qualified domain name
 
+ifdef::satellite[]
+* All {ProjectName} servers must be on the same version.
```
Comment (maximiliankolb) [inline guides/common/modules/ref_system-requirements.adoc:45]:
```suggestion
* All {ProjectServer}s must be on the same version.
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc [hunk 3]
```diff
@@ -26,6 +26,9 @@ For more information, see https://access.redhat.com/documentation/en-us/red_hat_
 * Ensure that you do not delete the manifest from the Customer Portal or in the {ProjectWebUI} because this removes all the entitlements of your content hosts.
 * Back up and remove all Foreman hooks before upgrading.
 Reinstate hooks only after {Project} is known to be working after the upgrade is complete.
+ifdef::satellite[]
+* All {ProjectName} servers must be on the same version.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:30]:
```suggestion
* All {ProjectServer}s must be on the same version.
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc [hunk 4]
```diff
@@ -50,6 +50,10 @@ For more information, see {AdministeringDocURL}Backing_Up_Server_and_Proxy_admin
 * Plan for updating any scripts you use that contain {Project} API commands because some API commands differ between versions of {Project}.
 For more information about changes in the API, see the Red Hat Knowledgebase article https://access.redhat.com/articles/4396911[API Changes Between {Project} Versions] on the Red{nbsp}Hat Customer Portal.
 
+ifndef::satellite[]
+Ensure that all {ProjectName} servers are on the same version.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:54]:
```suggestion
Ensure that all {ProjectServer}s are on the same version.
```

**Review by maximiliankolb — COMMENTED** [review]
I made some suggestions to use a different attribute.

File: guides/common/modules/proc_configuring-repositories.adoc [hunk 5]
```diff
@@ -2,14 +2,15 @@
...
+Use this procedure to enable the repositories that are required to install {ProductName}.
+Choose from the available list which operating system and version you are installing on:
 
 ifdef::foreman-el,katello[]
 * xref:#repositories-centos-8[CentOS 8]
 * xref:#repositories-centos-7[CentOS 7]
 endif::[]
 ifdef::foreman-el,katello,satellite[]
-* xref:#repositories-rhel-7[{RHEL} 7]
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
```
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/common/modules/proc_configuring-repositories.adoc:13]: This is not resolved yet.

File: guides/common/modules/proc_configuring-repositories.adoc [hunk 6]
```diff
@@ -2,14 +2,15 @@
...
+Use this procedure to enable the repositories that are required to install {ProductName}.
+Choose from the available list which operating system and version you are installing on:
 
 ifdef::foreman-el,katello[]
 * xref:#repositories-centos-8[CentOS 8]
 * xref:#repositories-centos-7[CentOS 7]
 endif::[]
 ifdef::foreman-el,katello,satellite[]
-* xref:#repositories-rhel-7[{RHEL} 7]
+* xref:#repositories-rhel-7[RHEL 7]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-repositories.adoc:13]: You are still making a change here that I don't see necessary. `{RHEL}` is an attribute expaning to "Red Hat Enterprise Linux": https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L90

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_configuring-repositories.adoc (hunk 6 above)
Comment (sabuchan07) [reply guides/common/modules/proc_configuring-repositories.adoc:13]: Sorry about that...that one was just me not thinking.

**Comment by sabuchan07** [comment]
@maximiliankolb Is this one ready to merge?

**Review by maximiliankolb — COMMENTED** [review]
Made some more suggestions. Please apply those, close the threads, and rebase to master. To craft your commit message, have a look at https://cbea.ms/git-commit/#seven-rules

File: guides/common/modules/proc_configuring-repositories.adoc [hunk 7]
```diff
@@ -2,15 +2,21 @@
...
 * xref:#repositories-centos-7[CentOS 7]
 endif::[]
 ifdef::foreman-el,katello,satellite[]
+<<<<<<< HEAD
 * xref:#repositories-rhel-8[{RHEL} 8]
 * xref:#repositories-rhel-7[{RHEL} 7]
+=======
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
+//* xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
+>>>>>>> 7b850960 (Added info about not supporting mixed version servers)
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-repositories.adoc:19]: This does not look correct. It looks like you had a merge conflict and did not correctly resolve it. `<<<<`, `====`, and `>>>>>` are markers from git which must not be commited.

File: guides/common/modules/ref_system-requirements.adoc [hunk 8]
```diff
@@ -41,9 +41,15 @@ endif::[]
 * A system umask of 0022
 * Full forward and reverse DNS resolution using a fully-qualified domain name
 
+<<<<<<< HEAD
 {ProjectServer} and {SmartProxyServer} do not support shortnames in the hostnames.
 When using custom certificates, the Common Name (CN) of the custom certificate must be a fully qualified domain name (FQDN) instead of a shortname.
 This does not apply to the clients of a {Project}.
+=======
```
Comment (maximiliankolb) [inline guides/common/modules/ref_system-requirements.adoc:48]: same as above.

File: guides/common/modules/proc_configuring-repositories.adoc [hunk 9]
```diff
@@ -2,15 +2,21 @@
...
 * xref:#repositories-centos-8[CentOS 8]
 * xref:#repositories-centos-7[CentOS 7]
 endif::[]
 ifdef::foreman-el,katello,satellite[]
+<<<<<<< HEAD
 * xref:#repositories-rhel-8[{RHEL} 8]
 * xref:#repositories-rhel-7[{RHEL} 7]
+=======
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
+//* xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-repositories.adoc:18]: Is this meant to be a comment? We generally don't add comments like this. Either we include it, or not.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Almost there, just some git markers left in there.

**Comment by maximiliankolb** [comment]
@sabuchan07 There are currently some merge conflicts. Please rebase to master and reword you commit message.

**Comment by sabuchan07** [comment]
@maximiliankolb Fixed the conflicts and rebased but, as you can clearly see, it's now added 250+ commits.  Just want you to know I'm aware of what happened, and I'll fix by squashing commits now.

**Comment by ekohl** [comment]
@sabuchan07 perhaps you rebased on the wrong branch? Forgot to fetch first? I'd try
```
# See which remote is theforeman/foreman-documentation
git remote -v
# Assuming it's orign
git fetch origin
git rebase origin/master
```

Then see which commits are left:
```
git log origin/master..HEAD
```

File: guides/common/modules/con_about-running-jobs-on-hosts.html [hunk 10]
```diff
@@ -0,0 +1,496 @@
+<!DOCTYPE html>
```
Comment (maximiliankolb) [inline guides/common/modules/con_about-running-jobs-on-hosts.html:1]: you'll want to remove this file.

**Comment by sabuchan07** [comment]
@ekohl Thanks for your suggestion!  I did all of that and, unfortunately, it's still listing a ton of commits.  Going to go about trying to remove them now.

**Comment by ekohl** [comment]
Ping me downstream if you want to go over this using screen sharing.

**Comment by sabuchan07** [comment]
Hi, @maximiliankolb !  Is this one ready to merge?  Thanks!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc [hunk 11]
```diff
@@ -38,6 +38,10 @@ For more information, see {AdministeringDocURL}Backing_Up_Server_and_Proxy_admin
 * Plan for updating any scripts you use that contain {Project} API commands because some API commands differ between versions of {Project}.
 For more information about changes in the API, see the Red Hat Knowledgebase article https://access.redhat.com/articles/4396911[API Changes Between {Project} Versions] on the Red{nbsp}Hat Customer Portal.
 
+ifndef::satellite[]
+Ensure that all {ProjectName} servers are on the same version.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:42]:
```suggestion
Ensure that all {ProjectServer}s are on the same version.
```

See https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L82

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc (hunk 3 above)
Comment (maximiliankolb) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:30]: This is not resolved yet.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
This needs some additional work.

**Comment by sabuchan07** [comment]
@maximiliankolb Apologies for missing those.  It should be all set now.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Please remove "guides/common/modules/con_about-running-jobs-on-hosts.html" from your branch.

please also rebase and squash your commit and reword your commit message.

**Comment by maximiliankolb** [comment]
Unfortunately your PR seems still broken; I can see a lot more changes in the diff than expected.

**Comment by sabuchan07** [comment]
@maximiliankolb This is all set now.

**Comment by maximiliankolb** [comment]
I think the change from "ProjectName servers" has been reverted; it should be "{ProjectServer}s".

**Comment by sabuchan07** [comment]
@maximiliankolb Fixed!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc [hunk 12]
```diff
@@ -38,6 +38,10 @@ For more information, see {AdministeringDocURL}Backing_Up_Server_and_Proxy_admin
 * Plan for updating any scripts you use that contain {Project} API commands because some API commands differ between versions of {Project}.
 For more information about changes in the API, see the Red Hat Knowledgebase article https://access.redhat.com/articles/4396911[API Changes Between {Project} Versions] on the Red{nbsp}Hat Customer Portal.
 
+ifndef::satellite[]
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:41]:
```suggestion
ifdef::satellite[]
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc (hunk 11 above)
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:42]: same as above.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc [hunk 13]
```diff
@@ -79,7 +79,7 @@ ifdef::katello,orcharhino,satellite[]
 . Upgrade to {project-client-name} on all content hosts.
 For more information, see xref:upgrading_content_hosts[].
 endif::[]
-ifndef::foreman-deb,orcharhino[]
+ifdef::foreman-deb,orcharhino[]
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:82]: please revert this change.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc [hunk 14]
```diff
@@ -57,7 +57,7 @@ You can upgrade to {ProjectName} {ProjectVersion} from {ProjectName} {ProjectVer
 For more information, see the https://access.redhat.com/documentation/en-us/red_hat_satellite/{ProjectVersionPrevious}/html/upgrading_and_updating_red_hat_satellite/[_{UpgradingDocTitle} to {ProjectVersionPrevious}_].
 endif::[]
 
-ifndef::satellite[]
+ifdef::satellite[]
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:60]: please revert this change.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

**Comment by sabuchan07** [comment]
@maximiliankolb Additional changes made

**Review by maximiliankolb — APPROVED** [review]
LGTM; I'll squash myself.

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: ✅ 
   ca5e7a591..c7382bdc3  3.4 -> 3.4
   ea5b39c14..8962ce391  3.3 -> 3.3
   55fade215..ace2f6827  3.2 -> 3.2
